Add docs about how to deploy Scalar products from AWS Marketplace (#184)
scalar-labs/scalar-kubernetes: docs/AwsMarketplaceGuide.md

# How to install Scalar products through AWS Marketplace

Scalar products (Scalar DB, Scalar DL, and their tools) are provided in AWS Marketplace as Container Image. This guide explains how to install Scalar products through AWS Marketplace.

Note that some Scalar products are licensed under commercial licenses, and the AWS Marketplace provides them as BYOL (Bring Your Own License). Please make sure you have appropriate licenses.

## Subscribe to Scalar products from AWS Marketplace

1. Access to the AWS Marketplace.
   * [Scalar DB (BYOL)]() // TODO: Add URL of Marketplace page after it's published
   * [Scalar DL Ledger (BYOL)]() // TODO: Add URL of Marketplace page after it's published
   * [Scalar DL Auditor (BYOL)]() // TODO: Add URL of Marketplace page after it's published

1. Select **Continue to Subscribe**.

1. Sign in to AWS Marketplace using your IAM user.  
   If you have already signed in, this step will be skipped automatically.

1. Read the **Terms and Conditions** and select **Accept Terms**.  

>    It takes some time. If it's done, you can see the current date in the **Effective date** column.

Reviewer: ```suggestion
   It takes some time. When it's done, you can see the current date in the **Effective date** column.
```

>    Also, you can see our product on the [Manage subscriptions](https://us-east-1.console.aws.amazon.com/marketplace/home#/subscriptions) page of AWS Console.

Reviewer: ```suggestion
   Also, you can see our products on the [Manage subscriptions](https://us-east-1.console.aws.amazon.com/marketplace/home#/subscriptions) page of AWS Console.
```

## **[BYOL]** Deploy containers on EKS (Amazon Elastic Kubernetes Service) from AWS Marketplace using Scalar Helm Charts

> If you subscribe to Scalar products in AWS Marketplace, you can pull container images of Scalar products from the private container registry ([ECR](https://aws.amazon.com/ecr/)) of AWS Marketplace. This section explains how to deploy Scalar products on your [EKS](https://aws.amazon.com/eks/) Cluster from the private container registry of AWS Marketplace.

Reviewer: ```suggestion
If you subscribe to Scalar products in AWS Marketplace, you can pull the container images of Scalar products from the private container registry ([ECR](https://aws.amazon.com/ecr/)) of AWS Marketplace. This section explains how to deploy Scalar products on your [EKS](https://aws.amazon.com/eks/) cluster from the private container registry.
```

1. Update the custom values file of the Helm Chart of a Scalar product you want to install.  
   You need to specify the private container registry (ECR) of AWS Marketplace and the version (tag) as the value of `[].image.repository` and `[].image.version (tag)` in the custom values file.  

>    * Examples

Reviewer: It looks too nested, so can you remove `Examples` and make it something like this?

* Scalar DB examples
* Scalar DL examples

Author: I updated it in 22d1432c1d0b1c987d47c4aa477e495982b749ed.

             ```
           * Scalar DL Auditor (scalardl-auditor-custom-values.yaml)
             ```yaml
             envoy:
               image:
                 repository: "709825985650.dkr.ecr.us-east-1.amazonaws.com/scalar/scalar-auditor-envoy"
                 version: "1.2.0"
             
             ...
             
             auditor:
               image:
                 repository: "709825985650.dkr.ecr.us-east-1.amazonaws.com/scalar/scalar-auditor"
                 version: "3.4.1"
             ```
           * Scalar DL Schema Loader for Ledger (schema-loader-ledger-custom-values.yaml)
             ```yaml
             schemaLoading:
               image:
                 repository: "709825985650.dkr.ecr.us-east-1.amazonaws.com/scalar/scalardl-schema-loader-ledger"
                 version: "3.4.1"
             ```
           * Scalar DL Schema Loader for Auditor (schema-loader-auditor-custom-values.yaml)
             ```yaml
             schemaLoading:
               image:
                 repository: "709825985650.dkr.ecr.us-east-1.amazonaws.com/scalar/scalardl-schema-loader-auditor"
                 version: "3.4.1"
             ```

> 1. Deploy the Scalar product using the Helm Chart with the above custom values file.

Reviewer: ```suggestion
1. Deploy the Scalar products using the Helm Chart with the above custom values files.
```

>    * Examples

Reviewer: Let's use the same structure as above for clarity.
(The above custom values files are kind of grouped into Scalar DB and Scalar DL, so let's make this also look the same)

Author: I updated it in 22d1432c1d0b1c987d47c4aa477e495982b749ed.

                 - name: "reg-ecr-mp-secrets"
             
             ...
             
             auditor:
               image:
                 repository: "709825985650.dkr.ecr.us-east-1.amazonaws.com/scalar/scalar-auditor"
                 version: "3.4.1"
               imagePullSecrets:
                 - name: "reg-ecr-mp-secrets"
             ```
           * Scalar DL Schema Loader for Ledger (schema-loader-ledger-custom-values.yaml)
             ```yaml
             schemaLoading:
               image:
                 repository: "709825985650.dkr.ecr.us-east-1.amazonaws.com/scalar/scalardl-schema-loader-ledger"
                 version: "3.4.1"
               imagePullSecrets:
                 - name: "reg-ecr-mp-secrets"
             ```
           * Scalar DL Schema Loader for Auditor (schema-loader-auditor-custom-values.yaml)
             ```yaml
             schemaLoading:
               image:
                 repository: "709825985650.dkr.ecr.us-east-1.amazonaws.com/scalar/scalardl-schema-loader-auditor"
                 version: "3.4.1"
               imagePullSecrets:
                 - name: "reg-ecr-mp-secrets"
             ```

> 1. Deploy the Scalar product using the Helm Chart with the above custom values file.

Reviewer: ```suggestion
1. Deploy the Scalar products using the Helm Chart with the above custom values files.
```
